### PR TITLE
[DSpace-CRIS] Nested / Basic Hierarchical Metadata (Frontend)

### PR DESCRIPTION
**Pull request by FrancescoMolinaro**
## References

Fixes: DSpace/DSpace#12159
Require backend: https://github.com/DSpace/DSpace/pull/11945

## Description

This PR replace the standard group in submission with nested metdatata and UI forms.
The default group in submission will become a nested form and will be opened in a modal, to isolate the form and create connections between pairs of metadata.

Example of configuration:

```
<row>
                <relation-field>
                    <relationship-type>isAuthorOfPublication</relationship-type>
                    <search-configuration>person</search-configuration>
                    <repeatable>true</repeatable>
                    <label>Author</label>
                    <hint>Enter the author's name (Family name, Given names).</hint>
                    <linked-metadata-field>
                        <dc-schema>dc</dc-schema>
                        <dc-element>contributor</dc-element>
                        <dc-qualifier>author</dc-qualifier>
                        <input-type>group</input-type>
                    </linked-metadata-field>
                    <externalsources>orcid</externalsources>
                    <required></required>
                    <!-- You may choose to validate author names via a Regular Expression if it's appropriate for
                         your institution. The below regex requires a comma to be present in the author field.
                         However, this is disabled by default to support organizations as authors, etc. -->
                    <!--<regex>\w+(,)+\w+</regex>-->
                </relation-field>
            </row>

 <form name="publicationStep-dc-contributor-author">
          <row>
            <field>
              <dc-schema>dc</dc-schema>
              <dc-element>contributor</dc-element>
              <dc-qualifier>author</dc-qualifier>
              <label>Author</label>
              <input-type>onebox</input-type>
              <repeatable>false</repeatable>
              <required>You must enter at least the author.</required>
              <hint>Enter the names of the authors of this item in the form Lastname, Firstname [i.e. Smith, Josh or Smith, J].</hint>
            </field>
          </row>
          <row>
            <field>
              <dc-schema>oairecerif</dc-schema>
              <dc-element>author</dc-element>
              <dc-qualifier>affiliation</dc-qualifier>
              <label>Affiliation</label>
              <input-type>onebox</input-type>
              <repeatable>false</repeatable>
              <required />
              <hint>Enter the affiliation of the author as stated on the publication.</hint>
            </field>
          </row>
</form>       
``` 
        
image

A new group type called inline-group has been added to mantain the groups inside the main form and not in a modal, follows and example of config:

**Please note that the inline-group as a limitation: all the fields needs to be configured inside one <row> tag.
If you want to achieve a display on multiple rows, it can be don through styling, adding for example to each field the style tag as follows:  <style>col-12</style>
This is a known limitation and should be added to the documentation as such.**

```

 <row>
              <field>
                <dc-schema>dc</dc-schema>
                <dc-element>contributor</dc-element>
                <dc-qualifier>editor</dc-qualifier>
                <label>Editor group</label>
                <input-type>inline-group</input-type>
                <repeatable>true</repeatable>
                <required />
                <hint>Add the editor of this publication</hint>
              </field>
</row>

<form name="publicationStep-dc-contributor-editor">
          <row>
            <field>
              <dc-schema>dc</dc-schema>
              <dc-element>contributor</dc-element>
              <dc-qualifier>editor</dc-qualifier>
              <label>Editor</label>
              <input-type>onebox</input-type>
              <repeatable>false</repeatable>
              <required>You must enter at least the author.</required>
              <hint>The editors of this publication.</hint>
            </field>
            <field>
              <dc-schema>oairecerif</dc-schema>
              <dc-element>editor</dc-element>
              <dc-qualifier>affiliation</dc-qualifier>
              <label>Affiliation</label>
              <input-type>onebox</input-type>
              <repeatable>false</repeatable>
              <required />
              <hint>Enter the affiliation of the editor as stated on the publication.</hint>
            </field>
          </row>
</form>
```

image

**Please note that is not possible to add repeatable fields inside the nested forms, this would result in errors in the UI breaking the form. This is a known limitation and should be added to the documentation as such.

Input type specific configuration, as the regex, should be added inside the nested form configuration, as they don't work on input types which represent nested forms. (not an actual control in the form) **

Note: A custom layout option will follow in a different PR, wich will allow to group those metadata also on Item page (CRIS Layout)

Note: For inline groups is possible to configure an additional button with the property showInlineGroupDuplicateButton , which if true, will display under the form a button to duplicate the whole section, metadata values included.

## Instructions for Reviewers

List of changes:

Added new inline-group field.
Adapted group field logic to use modal.
Adpated parsing and templates.

To test this feature follow the config and try to use the form groups in a submission.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You do not need to complete this checklist prior creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [ ] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [ ] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [ ] My PR **passes [ESLint](https://eslint.org/)** validation using `npm run lint`
- [ ] My PR **doesn't introduce circular dependencies** (verified via `npm run check-circ-deps`)
- [ ] My PR **includes [TypeDoc](https://typedoc.org/) comments** for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR **passes all specs/tests and includes new/updated specs or tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] My PR **aligns with [Accessibility guidelines](https://wiki.lyrasis.org/display/DSDOC9x/Accessibility)** if it makes changes to the user interface.
- [ ] My PR **uses i18n (internationalization) keys** instead of hardcoded English text, to allow for translations.
- [ ] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by MarieVerdonck** [comment]
**inline-group only renders first input & doesn't enforce required beyond name field**

The inline-group config only renders the first input of the affiliated nested form.
Screenshot 2026-03-13 at 04 48 17
Screenshot 2026-03-13 at 05 01 56
Screenshot 2026-03-13 at 05 03 14

Eg

```
        <form name="publicationStepGroup">
            <row>
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>contributor</dc-element>
                    <dc-qualifier>editor</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Editor</label>
                    <input-type>inline-group</input-type>
                    <hint>The editors of this publication.</hint>
                    <required>You must enter at least the editor's name.</required>
                </field>
            </row>
        </form>
<form name="publicationStepGroup-dc-contributor-editor">
    <row>
        <field>
            <dc-schema>dc</dc-schema>
            <dc-element>contributor</dc-element>
            <dc-qualifier>editor</dc-qualifier>
            <repeatable>false</repeatable>
            <label>Editor name</label>
            <input-type>onebox</input-type>
            <hint>Enter the names of the editor.</hint>
            <required>You must enter at least the editor.</required>
        </field>
    </row>
    <row>
        <field>
            <dc-schema>oairecerif</dc-schema>
            <dc-element>editor</dc-element>
            <dc-qualifier>affiliation</dc-qualifier>
            <repeatable>false</repeatable>
            <label>Affiliation</label>
            <input-type>onebox</input-type>
            <hint>Enter the affiliation of the editor.</hint>
            <required>Enter the affiliation of the editor.</required>
        </field>
    </row>
</form>
```

If switch Editor name and Affiliation inputs, only affiliation is shown.

Even with both required, I can submit form with only the name filled in!

**Comment by MarieVerdonck** [comment]
**Regex on nested form breaks input**

Adding regex on the author name field breaks input, can't add new authors, even when following the regex rules.

No validation of regex rule is shown.

Screenshot 2026-03-13 at 05 07 39
Screenshot 2026-03-13 at 05 08 15

Eg

```
        <form name="publicationStepGroup">
            <row>
                <relation-field>
                    <relationship-type>isAuthorOfPublication</relationship-type>
                    <search-configuration>person</search-configuration>
                    <repeatable>true</repeatable>
                    <label>Author</label>
                    <hint>Enter the author's name.</hint>
                    <linked-metadata-field>
                        <dc-schema>dc</dc-schema>
                        <dc-element>contributor</dc-element>
                        <dc-qualifier>author</dc-qualifier>
                        <input-type>group</input-type>
                    </linked-metadata-field>
                    <required>Enter at least one author</required>
                </relation-field>
            </row>
        </form>

        <form name="publicationStepGroup-dc-contributor-author">
            <row>
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>contributor</dc-element>
                    <dc-qualifier>author</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Author</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the names of the authors.</hint>
                    <required>You must enter at least the author.</required>
                    <regex>\w+(,)+\w+</regex>
                </field>
            </row>
            <row>
                <field>
                    <dc-schema>oairecerif</dc-schema>
                    <dc-element>author</dc-element>
                    <dc-qualifier>affiliation</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Affiliation</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the affiliation of the author.</hint>
                </field>
            </row>
        </form>
```

**Regex on top relation-field breaks form/validation**

Adding regex on the top relation-field input:

- can still add author name, not following regex
- if form saved/refreshed, nothing renders & error in console
Screenshot 2026-03-13 at 05 09 46
Screenshot 2026-03-13 at 05 10 08

Eg

```
        <form name="publicationStepGroup">
            <row>
                <relation-field>
                    <relationship-type>isAuthorOfPublication</relationship-type>
                    <search-configuration>person</search-configuration>
                    <repeatable>true</repeatable>
                    <label>Author</label>
                    <hint>Enter the author's name.</hint>
                    <linked-metadata-field>
                        <dc-schema>dc</dc-schema>
                        <dc-element>contributor</dc-element>
                        <dc-qualifier>author</dc-qualifier>
                        <input-type>group</input-type>
                    </linked-metadata-field>
                    <required>Enter at least one author</required>
                    <regex>\w+(,)+\w+</regex>
                </relation-field>
            </row>
        </form>

        <form name="publicationStepGroup-dc-contributor-author">
            <row>
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>contributor</dc-element>
                    <dc-qualifier>author</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Author</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the names of the authors.</hint>
                    <required>You must enter at least the author.</required>
                </field>
            </row>
            <row>
                <field>
                    <dc-schema>oairecerif</dc-schema>
                    <dc-element>author</dc-element>
                    <dc-qualifier>affiliation</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Affiliation</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the affiliation of the author.</hint>
                </field>
            </row>
        </form>
```

**Comment by MarieVerdonck** [comment]
**Putting repeatable=true on nested fields breaks form**

If making a nested input repeatable (eg affiliation of author) this breaks the form (can't add/edit new authors)
```
 ERROR TypeError: Cannot read properties of undefined (reading 'forEach')
    at dynamic-relation-group-modal.components.ts:231:22
    at Array.forEach (<anonymous>)
    at DsDynamicRelationGroupModalComponent.getMandatoryFieldModel (dynamic-relation-group-modal.components.ts:229:20)
    at DsDynamicRelationGroupModalComponent.getHeader (dynamic-relation-group-modal.components.ts:149:17)
    at DsDynamicRelationGroupModalComponent_Template (dynamic-relation-group-modal.component.html:3:29)
```
-> this happens because angular isn’t prepared to handle null values for these fields and there’s a general lack of null / undefined safety in DsDynamicRelationGroupModalComponent

Eg.

```
        <form name="publicationStepGroup">
            <row>
                <relation-field>
                    <relationship-type>isAuthorOfPublication</relationship-type>
                    <search-configuration>person</search-configuration>
                    <repeatable>true</repeatable>
                    <label>Author</label>
                    <hint>Enter the author's name.</hint>
                    <linked-metadata-field>
                        <dc-schema>dc</dc-schema>
                        <dc-element>contributor</dc-element>
                        <dc-qualifier>author</dc-qualifier>
                        <input-type>group</input-type>
                    </linked-metadata-field>
                    <required>Enter at least one author</required>
                </relation-field>
            </row>
        </form>

        <form name="publicationStepGroup-dc-contributor-author">
            <row>
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>contributor</dc-element>
                    <dc-qualifier>author</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Author</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the names of the authors.</hint>
                    <required>You must enter at least the author.</required>
                </field>
            </row>
            <row>
                <field>
                    <dc-schema>oairecerif</dc-schema>
                    <dc-element>author</dc-element>
                    <dc-qualifier>affiliation</dc-qualifier>
                    <repeatable>true</repeatable>
                    <label>Affiliation</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the affiliation of the author.</hint>
                </field>
            </row>
        </form>
```

Idem if adding repeatable = true on first (only visible) nested field of input-group input
Screenshot 2026-03-13 at 05 23 45

Eg
```
        <form name="publicationStepGroup">
            <row>
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>contributor</dc-element>
                    <dc-qualifier>author</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Author</label>
                    <input-type>inline-group</input-type>
                    <hint>The authors of this publication.</hint>
                    <required>You must enter at least the author's name.</required>
                </field>
            </row>
        </form>

        <form name="publicationStepGroup-dc-contributor-author">
            <row>
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>contributor</dc-element>
                    <dc-qualifier>author</dc-qualifier>
                    <repeatable>true</repeatable>
                    <label>Author</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the names of the authors.</hint>
                    <required>You must enter at least the author.</required>
                </field>
            </row>
            <row>
                <field>
                    <dc-schema>oairecerif</dc-schema>
                    <dc-element>author</dc-element>
                    <dc-qualifier>affiliation</dc-qualifier>
                    <repeatable>true</repeatable>
                    <label>Affiliation</label>
                    <input-type>onebox</input-type>
                    <hint>Enter the affiliation of the author.</hint>
                </field>
            </row>
        </form>
```

**Comment by FrancescoMolinaro** [comment]
Hi @MarieVerdonck, many thanks for the feedback, much appreciated.

Regarding the issue you reported with the repeatable fields, I have found out by checking the history of development on the CRIS code base, that the repeatable fields are not supported inside the nested forms.
I have update the description on the task, this information should be documented as a limitation of the feature in the documentation.

For the inline-group configuration there is an additional limitation, the form fields must be all inside a single row, if you want to achieve a layout on multiple rows, it can be done via the style attribute, I have updated the description accordingly.
Having the config inside a single row should also fix the issue whith the validation on multiple fields, e.g. multiple required fields should be validated separately.


About the regex issue instead I am not able to reproduce the problem with your configuration at https://github.com/DSpace/dspace-angular/pull/5097#issuecomment-4052493977 (regex on single field inside nested form).
I have used the same config and I see the the validation of the regex without being able to add the field, if it is not matching the regex:

[regex.webm](https://github.com/user-attachments/assets/5dabb7ec-de77-4de4-b721-36b96cd1ac8d)

Might I please ask you to try again? It could be possible that the problem has been mitigated after aligning the PR with the latest changes from main.

The second configuration instead is not possible, adding a regex on the top level metadata breaks the UI because is not an actual input, instead the regex should be configured on the fields inside the nested form.
I have updated the description accordingly.

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+      componentFixture.destroy();
+      testComponent = null;
+    });
+
+    it('should create DsDynamicRelationGroupModelComponent', () => {
+      expect(testComponent).toBeDefined();
+    });
+  });
+
+  xdescribe('when vocabulary has no external source option', () => {
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.component.spec.ts]: Why is this test section being skipped with `xdescribe`?

There's also a lot of commented out lines of code in this file altogether.  Are these tests working properly?

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+
+        expect(groupComp.chips.getChipsItems()).toEqual([]);
+        // groupComp.formCollapsed.subscribe((value) => {
+        //   expect(value).toEqual(false);
+        // });
+      });
+    });
+  });
+
+  xdescribe('when vocabulary has an external source option', () => {
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.component.spec.ts]: Again, this test block is being skipped with `xdescribe`

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+
+      it('should display external source button', inject([FormBuilderService], (service: FormBuilderService) => {
+        const testElement = debugElement.query(By.css('i.fa-share-square'));
+        expect(testElement).not.toBeNull();
+        expect(testElement.parent.nativeElement.disabled).toBeFalsy();
+      }));
+    });
+  });
+
+  xdescribe('when init model value has authority and submission scope is workflowitem', () => {
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.component.spec.ts]: And again here's another `xdescribe`

File: src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.components.ts [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+import { SubmissionService } from '../../../../../../../submission/submission.service';
+import { shrinkInOut } from '../../../../../../animations/shrink';
+import { FormComponent } from '../../../../../form.component';
+import { FormService } from '../../../../../form.service';
+import { FormBuilderService } from '../../../../form-builder.service';
+import { DsDynamicInputModel } from '../../ds-dynamic-input.model';
+import { DynamicRelationGroupModel } from '../dynamic-relation-group.model';
+
+/**
+ * Component representing a group input field
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/models/relation-group/modal/dynamic-relation-group-modal.components.ts]: Can we add some more comments to this larger component?  There's no comments on most of the methods of this component & the description here is very basic.  I think we should either have a better description of the behavior of this component *or* add more inline comments below to describe the various methods.

File: src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts [hunk 5]
```diff
@@ -516,4 +516,22 @@ export class DsDynamicFormControlContainerComponent extends DynamicFormControlCo
     this.subs.push(collection$.subscribe((collection) => this.collection = collection));
 
   }
+
+  isNotRequiredGroupAndEmpty(): boolean {
```
Comment (tdonohue) [inline src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts:524]: Could we add a basic description for this new method?

File: src/config/default-app-config.ts [hunk 6]
```diff
@@ -286,6 +286,7 @@ export class DefaultAppConfig implements AppConfig {
           },
         ],
       },
+      iconsVisibleWithNoAuthority: ['fas fa-user'],
```
Comment (tdonohue) [inline src/config/default-app-config.ts:292]: This new configuration doesn't appear to be described anywhere.  It should be added to `config.example.yml` with a description of what it does.

File: src/app/shared/form/form.component.ts [hunk 7]
```diff
@@ -368,11 +389,34 @@ export class FormComponent implements OnDestroy, OnInit {
...
+
+    this.formService.changeForm(this.formId, this.formModel);
+    if (!this.formBuilderService.isQualdropGroup(event.model as DynamicFormControlModel) && !this.isInlineGroupForm) {
+      // dispatch remove event for any field type except for qualdrop value and inline-group
       this.removeArrayItem.emit(event);
     }
-    this.formBuilderService.removeFormArrayGroup(index, formArrayControl, arrayContext);
+  }
+
+  clearScrollableDropdown($event, model: DynamicFormControlModel): void {
```
Comment (tdonohue) [inline src/app/shared/form/form.component.ts:475]: Could we add a basic description to all new methods in this class?

**Review by tdonohue — CHANGES_REQUESTED** [review]
@FrancescoMolinaro : Thanks for this PR.  I reviewed this and tested it with the backend PR.  Overall, the feature works well, and I haven't found any obvious bugs.  I did however find some minor things to cleanup in the code (especially some disabled tests -- see inline below).  

One other thing I did want to ask is whether the UI display for the `inline-group` is the same in this PR as in DSpace-CRIS?  I find the display to be a bit odd looking because there's so much vertical space used for each "inline group" of fields.  For example, here's what I'm seeing with the "Editor" field configured as an `inline-group`:

Image

Things that look odd to me include:
* The drag & drop icon is appearing *above* the field.  When this icon doesn't exist, it's empty space, so the field is always very tall.
* The delete button appears below the field, again using vertical space.
* I find the "Duplicate" and "+Add more" buttons to be a bit duplicative.  I did finally figure out though that they have different behavior.  I just find it odd to have both.

Overall, I expected this `inline-group` field to look a bit more like the existing `series` field (used for "Series/Report") where there delete button appears to the right of the fields & the drag & drop icon appears to the left of the fields.

In any case, this display issue is minor and we could move it to a follow-up ticket.  I just wanted to ask if this is the expected display or if it appears differently in DSpace-CRIS.

Overall, I'm nearly a +1 (once my inline comments are addressed/answered).  This all looks good to me. I do agree though that these new field types require more detailed **documentation**, especially since there's some invalid configurations that are possible (as pointed out by @MarieVerdonck in the above comments)

**Comment by tdonohue** [comment]
@MarieVerdonck : As we are nearing the deadline for 10.0 new features, I'd appreciate it if you can give this a followup review and explicitly note whether you are "+1" this being added to 10.0.  It sounds like your prior feedback has either been fixed or will be documented as an "invalid configuration".  I'll flag this PR as requiring docs now.

**Comment by FrancescoMolinaro** [comment]
Hi @tdonohue, many thanks for the review, I have updated the code adding comments and documentation where needed and adapting the group layout as you suggested.
For the duplicate button i have added a new configuration property to make it optional, showInlineGroupDuplicateButton, by default is false and the button won't be displayed.
I have also cleaned the group modal test suite, removing the tests that weren't necessary as they were part of a feature not yet included in the code,.
Hopefully this PR is now in a cleaner state, thanks again for the feedback.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @FrancescoMolinaro !  All my prior review feedback has been addressed.  I've also verified the new display for `inline-group` looks much better (& more compact):

Image

I also tested that setting `form > showInlineGroupDuplicateButton: true` will add the `Duplicate` buttons back into the display. 

So, this looks good to me. Thanks again for the hard work on this PR!

(Obviously this still will require documentation to be added to the DSDOC10x space for this feature.)

**Comment by tdonohue** [comment]
@MarieVerdonck : If you could give this PR another look, I think all your feedback has been addressed.  It also looks good to me now and I've officially given it a 👍    

I'd like to merge this quickly as several other merger-related PRs build off this one (so it will make it easier to code review those others PRs).  So, I plan to merge this tomorrow unless I hear from you sooner.

File: src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.html [hunk 8]
```diff
@@ -19,7 +19,7 @@
         <ng-container #componentViewContainer></ng-container>
       </div>
 
-      @if (hasHint && (formBuilderService.hasArrayGroupValue(model) || (!model.repeatable && (isRelationship === false || value?.value === null)) || (model.repeatable === true && context?.index === context?.context?.groups?.length - 1)) && (!showErrorMessages || errorMessages.length === 0)) {
+      @if (hasHint && (formBuilderService.hasArrayGroupValue(model) || ((!model.repeatable && (!(model?.isModelOfNotRepeatableGroup) || model?.isModelOfNotRepeatableGroup && context?.index === context?.context?.groups?.length - 1)) && (isRelationship === false || value?.value === null)) || (model.repeatable === true && context?.index === context?.context?.groups?.length - 1)) && (!showErrorMessages || errorMessages.length === 0)) {
```
Comment (MarieVerdonck) [inline src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.html]: Maybe move this logic into helper method, since not very readable/debuggable... (was already very long, now even longer)

File: src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.html [hunk 9]
```diff
@@ -28,7 +28,7 @@
         <div class="clearfix w-100 mb-2"></div>
       }
 
-      @if (!model.hideErrorMessages && showErrorMessages) {
+      @if (!model.hideErrorMessages && showErrorMessages && (!(model?.isModelOfNotRepeatableGroup) || model?.isModelOfNotRepeatableGroup && context?.index === context?.context?.groups?.length - 1) && isNotRequiredGroupAndEmpty()) {
```
Comment (MarieVerdonck) [inline src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.html]: Idem above => helper method to make code more understandable?

**Review by MarieVerdonck — APPROVED** [review]
Functionality seems to work as expected with adjustments to config & avoiding the repeatable/regex which is not allowed, added in REST review suggestion to do fail-fast parsing of submission forms config so can't build with invalid submission form config, but is not blocking issue.

Also tested `showInlineGroupDuplicateButton` => copies even authority controlled md, though I'm unsure what use case this functionality has, so user doesn't have to retype e.g. affiliation?

File: src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.html (hunk 9 above)
Comment (FrancescoMolinaro) [reply src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.html]: Hi @MarieVerdonck , you are right the template was starting to be unreadable, I refactored the logic checks into helper functions and documented them so that hopefully now are easier to read.

**Comment by tdonohue** [comment]
Thanks again @FrancescoMolinaro ! Merging as this is at +2.  Reminder that this has the "needs documentation" label until docs are added to https://wiki.lyrasis.org/display/DSDOC10x.

**Comment by GraziaQuercia** [comment]
documentation ready https://wiki.lyrasis.org/display/DSDOC10x/Nested+metadata+in+DSpace+items
